src/fswAlgorithms/orbitControl/lambertSolver/lambertSolver.cpp

```diff
@@ -0,0 +1,669 @@
+            // return no solution
+            return;
+        }
+        // for Gooding algorithm use halley root finder
+        std::array<double, 3> solution = halley(this->TOF, x0[0], this->numberOfRevolutions);
+        this->X = solution[0];
+        this->numIter = int(solution[1]);
+        this->errX = solution[2];
+
+        if (this->numberOfRevolutions > 0){
+            // if one or more orbits are to be completed, two solutions exist. Find second solution
+            std::array<double, 3> solution_sol2 = halley(this->TOF, x0[1], this->numberOfRevolutions);
+            this->XSol2 = solution_sol2[0];
+            this->numIterSol2 = int(solution_sol2[1]);
+            this->errXSol2 = solution_sol2[2];
+        }
+    }
+    else if (this->solverMethod == IZZO){
+        // get initial guess for iteration variable x
+        std::array<double, 2> x0 = izzoInitialGuess(this->lambda, this->TOF);
+        if (this->numberOfRevolutions > 0 && !this->multiRevSolution){
+            // if initial guess function determines that TOF is too short for multi-revolution solution to exist,
+            // return no solution
+            return;
+        }
+        // for Izzo algorithm use 3rd order householder root finder
+        std::array<double, 3> solution = householder(this->TOF, x0[0], this->numberOfRevolutions);
+        this->X = solution[0];
+        this->numIter = int(solution[1]);
+        this->errX = solution[2];
+
+        if (this->numberOfRevolutions > 0){
+            // if one or more orbits are to be completed, two solutions exist. Find second solution
+            std::array<double, 3> solution_sol2 = householder(this->TOF, x0[1], this->numberOfRevolutions);
+            this->XSol2 = solution_sol2[0];
+            this->numIterSol2 = int(solution_sol2[1]);
+            this->errXSol2 = solution_sol2[2];
+        }
+    }
+}
+
+/*! This method computes the velocities at the initial and final position for a given free variable x
+    @param x free variable of Lambert's problem that satisfies the given time of flight
+    @return std::array<Eigen::Vector3d, 2>
+*/
+std::array<Eigen::Vector3d, 2> LambertSolver::computeVelocities(double x)
+{
+    double c = (this->r2vec - this->r1vec).norm(); // chord length
+    double r1 = this->r1vec.norm();
+    double r2 = this->r2vec.norm();
+    double s = 1.0/2.0*(r1+r2+c); // semiperimeter
+
+    double y = sqrt(1.0-pow(this->lambda,2)*(1.0-pow(x,2)));
+    double gamma = sqrt(this->mu * s/2);
+    double rho = (r1-r2)/c;
+    double sigma = sqrt(1.0-pow(rho,2));
+
+    // compute velocity components for v1vec and v2vec
+    double Vr1 = gamma*((this->lambda*y - x) - rho*(this->lambda*y+x))/r1;
+    double Vr2 = -gamma*((this->lambda*y - x) + rho*(this->lambda*y+x))/r2;
+    double Vt1 = gamma*sigma*(y+this->lambda*x)/r1;
+    double Vt2 = gamma*sigma*(y+this->lambda*x)/r2;
+
+    // get orbit frame unit vectors
+    Eigen::Vector3d i_r1 = this->Oframe1.at(0);
```

Comment:
add frame denotation to all matrix representations of vectors

Reply:
Done.